src/formatDate.js

```diff
@@ -50,7 +50,59 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const firstDivider = fromFormat[3];
+  const secondDivider = toFormat[3];
+  const originalDate = date.split(firstDivider);
+
+  const reverseDate = [];
+  const dateObj = {
+
+  };
+
+  for (let i = 0; i < 3; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        dateObj.DD = originalDate[i];
+        break;
+
+      case 'MM':
+        dateObj.MM = originalDate[i];
+        break;
+
+      case 'YYYY':
+        dateObj.YYYY = originalDate[i];
+        break;
+
+      case 'YY':
+        if (+originalDate[i] < 30) {
+          dateObj.YYYY = 20 + originalDate[i];
+        } else {
+          dateObj.YYYY = 19 + originalDate[i];
+        }
+        break;
+    }
+  }
+
+  for (let i = 0; i < 3; i++) {
+    switch (toFormat[i]) {
+      case 'DD':
+        reverseDate.push(dateObj.DD);
+        break;
+      case 'MM':
+        reverseDate.push(dateObj.MM);
+        break;
+      case 'YYYY':
+        reverseDate.push(dateObj.YYYY);
+        break;
+      case 'YY':
+        reverseDate[i] = dateObj.YYYY.slice(2);
```

Comment:
and rewrite logic here
toFormat[i] - is a key of your date object

Reply:
One more PR. Spent a lot of time to complete it though can`t get rid of those if-blocks. How else can I manipulate the 'year' fotmat!? Anyway, this is my final version for now.